### PR TITLE
Sysreg

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Comment by igoropaniuk** [comment]
`Tested-by: Igor Opaniuk <igor.opaniuk@linaro.org>`

minor: btw, for some reason, while generating `arm32_gicv3_sysreg.S`  and 
comment is empty, it still adds `#` but with trailing space

but this check in `scripts/arm32_sysreg.py` should handle this case:
```
        if len(comment) == 0:
            print('')
            return
        if s_file:
            print('# ' + comment)
        else:
            print('/* ' + comment + ' */')
```

selection_0![selection_02626](https://user-images.githubusercontent.com/22025908/44335486-40cf6100-a47d-11e8-92df-7d4804060f89.png)

**Comment by jenswi-linaro** [comment]
Update

**Comment by jbech-linaro** [comment]
Not sure why IBART fails during flash, trying to debug. In any case, it's not caused by the changes in this PR.

**Comment by jbech-linaro** [comment]
IBART OK, the time default timeout (3 seconds) was hit when trying going to recovery mode. I raised it to 10 seconds and now it seemed to pass.

**Comment by jforissier** [comment]
Shippable CI has found an issue with `PLATFORM=stm32mp1`...

**Comment by jenswi-linaro** [comment]
Update with "core: rename read_idpfr1() to read_id_pfr1() " to fix the build error. Also squashed and rebased get the right order of the commits.

File: core/arch/arm/arm.mk [hunk 1]
```diff
@@ -208,6 +208,11 @@ arm32-sysregs-$(arm32-sysreg-txt)-h := arm32_sysreg.h
 arm32-sysregs-$(arm32-sysreg-txt)-s := arm32_sysreg.S
 arm32-sysregs += $(arm32-sysreg-txt)
 
+arm32-gicv3-sysreg-txt = core/arch/arm/kernel/arm32_gicv3_sysreg.txt
```
Comment (jforissier) [inline core/arch/arm/arm.mk]: `ifeq ($(CFG_ARM_GICV3),y) ... endif`?

Comment (jenswi-linaro) [reply core/arch/arm/arm.mk]: OK

**Comment by jenswi-linaro** [comment]
Comment addressed, pep8 is now happy, tag applied

**Pull request merged**